memory-viz/src/memory_model.ts

```diff
@@ -167,39 +176,41 @@ export class MemoryModel {
      * the SVG documentation. For the styling options in terms of boxes, refer to the Rough.js documentation.
      */
     drawObject(
-        x: number,
-        y: number,
-        type: string,
-        id: number,
+        x: number | undefined = 0,
+        y: number | undefined = 0,
+        type: string | undefined,
+        id: number | undefined = 0,
```

Comment:
It looks like you misunderstood my previous comment. My comments always refer to one specific line (unless I specify otherwise, or select a range of lines).

The only parameters I wanted you to make optional were `id` and `show_indexes`. For `id`, do not specify a default value; the code itself should handle when `id === undefined`.